h2o-core/src/test/java/water/TestUtil.java

```diff
@@ -90,17 +90,30 @@ public static void checkLeakedKeys() {
             (value.isJob() && value.<Job>get().isStopped()) ) {
           leaked_keys--;
         } else {
-          System.out.println(k + " -> " + value.get());
+          try {
+            System.out.println(k + " -> " + value.get());
+          } catch(Exception x) {
+            System.out.println(k + " -> " + value);
+          }
           if( cnt++ < 10 )
             System.err.println("Leaked key: " + k + " = " + TypeMap.className(value.type()));
         }
       }
       if( 10 < leaked_keys ) System.err.println("... and "+(leaked_keys-10)+" more leaked keys");
     }
     assertTrue("Keys leaked: " + leaked_keys + ", cnt = " + cnt, leaked_keys <= 0 || cnt == 0);
-    // Bulk brainless key removal.  Completely wipes all Keys without regard.
+    removeKeysRegardless();
+
+  }
+
+  public static int numberOfLeakedKeys() {
+    return H2O.store_size() - _initial_keycnt;
+  }
+
+  // Bulk brainless key removal.  Completely wipes all Keys without regard.
+  public static void removeKeysRegardless() {
```

Comment:
This is dangerous - the tests need to fail if there is a leak. 
This allows people to hide leaks.